cloudinit/sources/DataSourceAzure.py

```diff
@@ -513,9 +515,10 @@ def crawl_metadata(self):
 
         if perform_reprovision or perform_reprovision_after_nic_attach:
             LOG.info("Reporting ready to Azure after getting ReprovisionData")
-            use_cached_ephemeral = self.distro.networking.is_up(
-                self.fallback_interface
-            ) and getattr(self, "_ephemeral_dhcp_ctx", None)
+            use_cached_ephemeral = (
+                self.distro.networking.is_up(self.fallback_interface)
+                and self._ephemeral_dhcp_ctx
```

Comment:
Do we need to check for self._ephemeral_dhcp_ctx.lease here?

Reply:
In practice I expect that a NoDHCPLeaseError exception would have been raised and prevented it from getting this far.

I added an explicit check and I have other PRs in queue to further improve this handling.